enhancements/sig-architecture/225-clusteradm-operator/README.md

```diff
@@ -0,0 +1,557 @@
+
+### Non-goals
+
+- Extending or modifying the core OCM functionality in any way
+
+## Proposal
+
+Add a new component to the OCM ecosystem: the clusteradm operator. It will reconcile a new OCM `MultiCluster` CRD, which encapsulates all information required to bootstrap and maintain an OCM multi-cluster. The spec of a `MultiCluster` will contain all the same information traditionally provided to `clusteradm init` and `clusteradm join`. The `MultiCluster` API surface will feature 1:1 parity with the `clusteradm` CLI, including all optional flags. In some cases, e.g., kubeconfigs, the spec will contain references to secrets containing auxiliary data to avoid exposing sensitive information.
+
+### User Stories
+
+#### Story 1
+
+As an infrastructure or platform engineer, I want to apply a custom resource to a Kubernetes cluster and have it bootstrap itself (plus zero or more additional Kubernetes clusters) into an OCM multi-cluster, so that I do not have to manually orchestrate a series of imperative CLI commands.
+
+#### Story 2
+
+As an infrastructure or platform engineer, I want to commit the definition of my OCM multi-cluster into version control, so that I can gate changes to it using code review and leverage a GitOps approach for managing its lifecycle and topology.
+
+### Risks and Mitigation
+
+1. People may not want to store kubeconfigs for their spoke clusters on the hub cluster. Especially in a non-EKS scenario where the kubeconfigs contain long lived tokens. This can be mitigated by continuing to expand OCM's workload identify capabilities to support native integration with Azure and GCP in alignment with security best practices. Spoke kubeconfigs _could_ also be deleted from the hub cluster following the initial bootstrapping of the multi-cluster, but that introduces many challeges regarding upgrades, reconfiguration, etc.
+
+1. In certain environments, network connectivity from hub to spoke cannot be assumed. The clusteradm operator will **not** be able to bootstrap such spoke clusters automatically. However, additional metadata can be provided when these clusters are joined via `clusteradm join ... --klusterlet-annotation foo=bar`. The clusteradm operator on the hub can subsequently identify spoke clusters that were bootstrapped out-of-band and incorporate the health status of their corresponding `ManagedCluster` objects into the health of the overall `MultiCluster`.
+
+## Design Details
+
+### API change
+
+No changes to the core OCM APIs are required.
+
+### MultiCluster API
+
+Current state of the v1alpha1 `MultiCluster` API, which is fully functional and meets all the capabilities outlined elsewhere in this enhancement proposal.
+
+```go
+package v1alpha1
+
+import (
+	"fmt"
+	"sort"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// MultiClusterSpec defines the desired state of MultiCluster.
+type MultiClusterSpec struct {
+	Hub              Hub               `json:"hub"`
```

Comment:
is this necessary? I think user can just apply ClusterManager resource.

Reply:
That assumes that the OCM hub operators are already installed to reconcile the `ClusterManager`. How would the OCM hub components be initialized in the scenario you're describing?

By running `clusteradm init`, the OCM hub components and installed _and_ the `ClusterManager` is created.